tools/pipelines/templates/build-npm-client-package.yml

```diff
@@ -0,0 +1,898 @@
+
+                - task: DownloadPipelineArtifact@2
+                  inputs:
+                    artifact: nested_lib_dist
+                    targetPath: $(Build.SourcesDirectory)
+
+                - script: |
+                    echo "Extracting nested lib/dist directories and their contents..."
+                    tar --extract --gzip --file $(Build.SourcesDirectory)/nested_lib_dist.tar.gz --directory $(Build.SourcesDirectory)
+                  displayName: Extract Nested Lib/Dist Directories and Their Contents
+
+                # Test
+
+                # Set variable startTest if the build succeed so that we can run all the test tasks whether they are failed or not
+                - script: |
+                    echo "##vso[task.setvariable variable=startTest]true"
+                  displayName: Start Test
+
+                - ${{ if contains(test.name, 'jest') }}:
+                  - task: Npm@1
+                    displayName: 'Run build for jest tests'
+                    inputs:
+                      command: custom
+                      workingDir: '${{ parameters.buildDirectory }}'
+                      customCommand: 'run ci:build'
+
+                - template: /tools/pipelines/templates/include-test-task.yml@self
+                  parameters:
+                    taskTestStep: '${{ test.name }}'
+                    buildDirectory: '${{ parameters.buildDirectory }}'
+
+                - task: Npm@1
+                  displayName: 'npm run test:copyresults'
+                  inputs:
+                    command: custom
+                    workingDir: '${{ parameters.buildDirectory }}'
+                    customCommand: 'run test:copyresults'
+
+                - ${{ if contains(test.name, 'tinylicious') }}:
+                  - task: Bash@3
+                    displayName: Upload tinylicious log
+                    condition: always()
+                    continueOnError: true # Keep running subsequent tasks even if this one fails (e.g. the tinylicious log wasn't there)
+                    inputs:
+                      targetType: inline
+                      script: |
+                        set -eu -o pipefail
+                        PATH_TO_TINYLICIOUS_LOG=$(Build.SourcesDirectory)/packages/test/test-end-to-end-tests/tinylicious.log;
+                        if [ -f $PATH_TO_TINYLICIOUS_LOG ] ; then
+                          echo "Found file at '$PATH_TO_TINYLICIOUS_LOG'. Uploading.";
+                          echo "##vso[task.uploadfile]$PATH_TO_TINYLICIOUS_LOG";
+                        else
+                          echo "##vso[task.logissue type=warning]Failed to upload tinylicious log file ('$PATH_TO_TINYLICIOUS_LOG' not found).";
+                        fi
+
+                  # At this point we want to publish the tinylicious.log artifact, but as part of 1ES migration
+                  # that is now part of templateContext.outputs below.
+
```

Comment:
Outdated comment, this should have been removed from main a while back. We don't publish that log as a pipeline artifact anymore, but as a pipeline _log file_ by using `##vso[task.uploadfile]$PATH_TO_TINYLICIOUS_LOG`.
```suggestion
```